subxt/src/backend/unstable/follow_stream_unpin.rs

```diff
@@ -117,11 +118,15 @@ impl<Hash: BlockHash> Stream for FollowStreamUnpin<Hash> {
                     let rel_block_num = this.rel_block_num;
                     // Pin this block, but note that it can be unpinned any time since it won't show up again (except
                     // as a parent block, which we are ignoring at the moment).
-                    let block_ref =
-                        this.pin_unpinnable_block_at(rel_block_num, details.finalized_block_hash);
+
+                    let finalized_block_hashes = details
+                        .finalized_block_hashes
+                        .iter()
+                        .map(|h| this.pin_unpinnable_block_at(rel_block_num, *h))
```

Comment:
I believe that `rel_block_num` should be incremented for each finalized block, so that the oldest one is unpinned first etc.